homeassistant/components/assist_pipeline/__init__.py

```diff
@@ -130,3 +131,19 @@ async def async_pipeline_from_audio_stream(
     )
     await pipeline_input.validate()
     await pipeline_input.execute()
+
+
+async def async_audio_stream_from_queue(
```

Comment:
feels a bit like a generic util function ? queue to generator with optional timeout ?

Reply:
Somewhat, though it is specific to `bytes` as queue items. I guess that could be made generic.

Reply:
Moved to `util.async_`